mriqc/config.py

```diff
@@ -405,24 +405,41 @@ def init(cls):
         """Create a new BIDS Layout accessible with :attr:`~execution.layout`."""
         if cls._layout is None:
             import re
-
+            from bids.layout.index import BIDSLayoutIndexer
             from bids.layout import BIDSLayout
 
-            work_dir = cls.work_dir / "bids.db"
-            work_dir.mkdir(exist_ok=True, parents=True)
-            cls._layout = BIDSLayout(
-                str(cls.bids_dir),
+            _db_path = cls.bids_database_dir or (
+                cls.work_dir / "bids_db"
+            )
+            _db_path.mkdir(exist_ok=True, parents=True)
+
+            # Recommended after PyBIDS 12.1
+            _indexer = BIDSLayoutIndexer(
                 validate=False,
-                # database_path=str(work_dir),
                 ignore=(
                     "code",
                     "stimuli",
                     "sourcedata",
                     "models",
                     "derivatives",
+                    "scripts",
                     re.compile(r"^\."),
+                    re.compile(
+                        r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(fmap|dwi)/"
+                    ),
+                    re.compile(
+                        r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(beh|dwi|eeg|ieeg|meg|perf)"
+                    ),
```

Comment:
I assume ignore will apply to the basenames, not the full name. Have you checked whether these can just be listed as `["fmap", "dwi", "beh", ...]`?

Reply:
will check

Reply:
I have checked and modality folders ("dwi", "fmap") cannot be just listed. Testing with "dwi" also showed it doesn't match the suffix either.

I went ahead and added regexps for all the modalities in the entity table that one could posibly have. Would it make sense for PyBIDS to expose an argument of `include_patterns` (just `include`?) the same way excluded patterns are exposed?

Reply:
I believe there's `force_index` that's the equivalent of `include`.

Reply:
Okay, I followed a bit in the code and indeed `force_index` seems to be the argument, and it does have prevalence over the exclude patterns. But it also seems we'll need to add an `ignore` regexp with `/.*/` for the accept-list approach (instead of a deny list) to work.

Reply:
Okay, I have just tried with MyConnectome the following:
```Python
indexer = BIDSLayoutIndexer(
    validate=False,
    ignore=("sourcedata", re.compile(r".*"), ),
    force_index=(
        re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/anat/.*_T(1|2)w\."),
        re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/func/.*_bold\."),
    ),
)
```

and 1) sourcedata keeps being indexed (it matches the force_index); 2) it takes forever to initialize the layout, much more than the deny list approach.

btw inserting `^` in the begining makes it stop matching anything.